tests/datasource/data_connector/test_data_connector.py

```diff
@@ -112,11 +112,8 @@ def test__file_object_caching_for_FileDataConnector(tmp_path_factory):
         assets={"stuff": {}},
     )
 
-    with pytest.raises(ValueError):
-        my_data_connector.get_data_reference_list_count()
-
-    with pytest.raises(ValueError):
-        my_data_connector.get_unmatched_data_references()
+    assert my_data_connector.get_data_reference_list_count() == 0
+    assert my_data_connector.get_unmatched_data_references() == []
```

Comment:
There's a very similar assertion on 121 - I would either change this one or that one to parallel each other
```suggestion
    assert len(my_data_connector.get_unmatched_data_references()) == 0
```